.github/workflows/node.js.yml

```diff
@@ -11,22 +14,66 @@ on:
 
 jobs:
   build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [21.x]
+
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node-version }}
+        cache: 'npm'
+
+    - name: Install dependencies
+      run: npm ci
+
+    - name: Build application
+      run: npm run build --if-present
 
+    - name: Upload build artifacts
+      uses: actions/upload-artifact@v4
+      with:
+        name: build-files
+        path: |
+          .next/
+          node_modules/
+        retention-days: 1
+
+  test:
     runs-on: ubuntu-latest
+    needs: build
 
     strategy:
       matrix:
         node-version: [21.x]
-        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
 
     steps:
     - uses: actions/checkout@v4
+
     - name: Use Node.js ${{ matrix.node-version }}
       uses: actions/setup-node@v3
       with:
         node-version: ${{ matrix.node-version }}
         cache: 'npm'
-    - run: npm ci
-    - run: npm run build --if-present
-    # TODO: once we have a test runner/framework in place, uncomment this
-    # - run: npm test
+
+    - name: Download build artifacts
```

Comment:
Where is this downloaded to and what are we using it for? (or my own education)

Reply:
That's no longer part of the build process since we're not trying to cache anything in between a multi-version matrix-strategy build process.